### PR TITLE
TopMaliciousRatioIndicators: Fix error when no indicators were found

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/43070)

## Description
- Fixed an issue where the script failed to run if no indicators were found.

## Screenshots
Before (error) and after fix for the same command execution:
Screen Shot 2021-11-21 at 18 33 59

### CODE REVIEW

**Comment by Bargenish** [comment]
Validation fails due to ST106 -` You should update the following file to the package format`

**Review by yaakovi — COMMENTED** [review]
please convert to package format and add a test.
this is a widget script, how does it look like as a widget?

**Comment by Bargenish** [comment]
Widget without data(fixed error)
Screen Shot 2021-11-23 at 22 55 01

Widget with data:
Screen Shot 2021-11-23 at 22 54 55

File: Packs/CommonScripts/Scripts/TopMaliciousRatioIndicators/TopMaliciousRatioIndicators.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    indicators = res[0]['Contents']
+    indicators = [i for i in indicators if len(i.get('investigationIDs') or []) >= min_number_of_invs]
+
+    if not indicators:
+        return json.dumps({"total": 0, "data": []}), []
+
+    indicators_map = {i['id']: i for i in indicators}
+
+    res = demisto.executeCommand("maliciousRatio", {'id': ",".join(indicators_map)})
+    malicious_ratio_result = res[0]['Contents']
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/TopMaliciousRatioIndicators/TopMaliciousRatioIndicators.py:40]:
```suggestion
    malicious_ratio_result = execute_command("maliciousRatio", {'id': ",".join(indicators_map)})
```

File: Packs/CommonScripts/Scripts/TopMaliciousRatioIndicators/TopMaliciousRatioIndicators.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if value not in exist_values:
+            exist_values.add(value)
+            result.append(e)
+    return result
+
+
+def find_indicators_with_mal_ratio(
+        max_indicators: int, min_number_of_invs: int, max_results: int, from_date: str) -> Tuple[str, list]:
+    res = demisto.executeCommand("findIndicators", {'query': f'lastSeen:>={from_date}', 'size': max_indicators})
+    indicators = res[0]['Contents']
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/TopMaliciousRatioIndicators/TopMaliciousRatioIndicators.py:31]:
```suggestion
    indicators = execute_command("findIndicators", {'query': f'lastSeen:>={from_date}', 'size': max_indicators})
```

File: Packs/CommonScripts/Scripts/TopMaliciousRatioIndicators/TopMaliciousRatioIndicators.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+
+def select_indicator_columns(indicator: dict) -> Dict:
+    display_indicator = {}
+    display_indicator['ID'] = indicator['id']
+    display_indicator['Type'] = indicator['indicator_type']
+    display_indicator['Malicious Ratio'] = '%.2f' % float(indicator['maliciousRatio'])
+    display_indicator['Value'] = indicator['value']
+    display_indicator['Last Seen'] = indicator['lastSeen']
+    return display_indicator
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/TopMaliciousRatioIndicators/TopMaliciousRatioIndicators.py:14]:
```suggestion
    return {
        'ID': indicator['id'],
        'Type': indicator['indicator_type'],
        'Malicious Ratio': '%.2f' % float(indicator['maliciousRatio']),
        'Value': indicator['value'],
        'Last Seen': indicator['lastSeen'],
    }
```

File: Packs/CommonScripts/Scripts/TopMaliciousRatioIndicators/TopMaliciousRatioIndicators.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+
+def dedup_by_value(indicators_list: list) -> List:
+    exist_values = set()
+    result = []
+    for e in indicators_list:
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/TopMaliciousRatioIndicators/TopMaliciousRatioIndicators.py:20]:
```suggestion
def dedup_by_value(indicators: list) -> List:
    exist_values = set()
    result = []
    for e in indicators:
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**